invoicex/reader/ttrack.py

```diff
@@ -91,21 +82,19 @@ def _group_tasks_remove_duplicates(self, v):
     def _group_time_and_sum(self, v):
         return v.sum()
 
-    def _group_tasks_and_time(self):
-        df = self._filter_by_month(
-            "2022-06"
-        )  # TODO Externalize _filter_by_month to the same function that _get_query()
-        print(df)
+    def _generate_dataframe(self):
+        """Create the final DataFrame"""
+        df = self._filter_by_month(self.year_month)
         df_grouped = df.groupby("date").aggregate(
             lambda v: self._group_tasks_remove_duplicates(v)
             if v.name == "task"
             else self._group_time_and_sum(v)
         )
         return df_grouped
 
-    def __call__(self, *args: Any, **kwds: Any) -> Any:
-        return self._group_tasks_and_time()
-
 
-# e = TTrack(TTRACK_DB)
-# print(e())
+def get_data(args) -> pd.DataFrame:
```

Comment:
keep in mind that it would be just temporary because ComposeReader will instantiate TTrack inside (probably an attribute called ttrack inside the ComposeReader class)

Reply:
I'll handle all the comments in next commits, sending now what I got while trying to async TTrack to see if it is correct before externalizing it in ComposeReader